src/Param.cc

```diff
@@ -428,7 +430,122 @@ bool ParseColorUsingStringStream(const std::string &_input,
+/// (expects 6 or 7 values) and the resulting pose is valid.
+/// \param[in] _input Input string.
+/// \param[in] _key Key of the parameter, used for error message.
+/// \param[in] _attributes Attributes associated to this pose.
+/// \param[out] _value This will be set with the parsed value.
+/// \return True if parsing pose succeeded.
+bool ParsePoseUsingStringStream(const std::string &_input,
+    const std::string &_key, const Param_V &_attributes,
+    ParamPrivate::ParamVariant &_value)
+{
+  StringStreamClassicLocale ss(_input);
+  std::string token;
+  std::vector<double> values;
+  double v;
+  bool isValidPose = true;
+  while (ss >> token)
+  {
+    try
+    {
+      v = std::stod(token);
+    }
+    // Catch invalid argument exception from std::stod
+    catch(std::invalid_argument &)
+    {
+      sdferr << "Invalid argument. Unable to set value ["<< token
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+    // Catch out of range exception from std::stof
+    catch(std::out_of_range &)
+    {
+      sdferr << "Out of range. Unable to set value [" << token
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+
+    if (!std::isfinite(v))
+    {
+      sdferr << "Pose values must be finite.\n";
+      isValidPose = false;
+      break;
+    }
+
+    values.push_back(v);
+  }
+
+  if (!isValidPose)
+    return false;
+
+  const std::string defaultRotationType = "rpy_radians";
+  std::string rotationType = defaultRotationType;
+  for (const auto &p : _attributes)
+  {
+    if (p->GetKey() == "rotation_type")
+      rotationType = p->GetAsString();
+  }
+
+  std::string desiredSizeErrorMessage;
+  auto isRotationType =
+      [&](const std::string &_rotationType, std::size_t _desiredSize)
+  {
+    if (rotationType != _rotationType)
+    {
+      return false;
+    }
+    else if (values.size() != _desiredSize)
+    {
+      std::stringstream sstream;
+      sstream << "The value for //pose[@rotation_type='" << _rotationType
+          << "'] must have " << _desiredSize << " values, but " << values.size()
+          << " were found instead.\n";
+      desiredSizeErrorMessage = sstream.str();
+      return false;
+    }
+    return true;
+  };
+
+  if (isRotationType("rpy_radians", 6u))
+  {
+    _value = ignition::math::Pose3d(values[0], values[1], values[2],
+        values[3], values[4], values[5]);
+  }
+  else if (isRotationType("rpy_degrees", 6u))
+  {
+    _value = ignition::math::Pose3d(values[0], values[1], values[2],
+        IGN_DTOR(values[3]), IGN_DTOR(values[4]), IGN_DTOR(values[5]));
+  }
+  else if (isRotationType("q_wxyz", 7u))
+  {
+    _value = ignition::math::Pose3d(values[0], values[1], values[2],
+        values[3], values[4], values[5], values[6]);
+  }
+  else
+  {
+    if (desiredSizeErrorMessage.empty())
+    {
+      sdferr << "Invalid @rotation_type of [" << rotationType
```

Comment:
nit Above, you include single quotes around the value of `@rotation_type`; however, here, you mix between surrounding with braces `[]` or no quotes.

Consider surrounding all values (actual and supported) with single quotes for consistency.

Reply:
Totally missed it, modified in https://github.com/ignitionrobotics/sdformat/pull/589/commits/8c776955e9d39d25d38acbde24434f3b422156e8, thanks!